### PR TITLE
acceptance tests in docker

### PR DESCRIPTION
**Pull request by wkloucek**
Acceptance tests can now be run inside docker containers on the developers machine. So no more need to install php and php system dependencies. Docs how to run: https://github.com/owncloud/ocis/blob/acceptance_tests_in_docker/docs/ocis/development/testing.md#testing-with-test-suite-in-docker

The root makefile now attempts to guide the user through the repository:
```
Please use 'make <target>' where <target> is one of the following:

Testing with test suite natively installed:

	docs: https://owncloud.github.io/ocis/development/testing/#testing-with-test-suite-natively-installed

	make test-acceptance-api	run API acceptance tests
	make clean-tests		delete API tests framework dependencies

---------------------------------------------------------

You also should have a look at other available Makefiles:

oCIS:

	docs: https://owncloud.github.io/ocis/development/building/

	see ./ocis/Makefile
	or run make -C ocis help

Documentation:

	docs: https://owncloud.github.io/ocis/development/building-docs/

	see ./docs/Makefile
	or run make -C docs help

Testing with test suite in docker:

	docs: https://owncloud.github.io/ocis/development/testing/#testing-with-test-suite-in-docker

	see ./tests/acceptance/docker/Makefile
	or run make -C tests/acceptance/docker help
```

### CODE REVIEW

**Comment by micbar** [comment]
Wonderful!!

**Comment by wkloucek** [comment]
> Wonderful!!

Did you try it work for you? Could you do a review / accept the PR?

File: docs/ocis/development/testing.md [hunk 1]
```diff
@@ -9,7 +9,79 @@ geekdocFilePath: testing.md
...
+For example `make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis` runs the same tests as the `localApiTests-apiOcisSpecific-ocis` CI pipeline, which runs the oCIS test suite "apiOcisSpecific" against an oCIS with oCIS storage.
+
+For example `make -C tests/acceptance/docker Core-API-Tests-owncloud-storage-3`runs the same tests as the `Core-API-Tests-owncloud-storage-3` CI pipline, which runs the third (out of ten) ownCloud test suite against an oCIS with owncloud storage.
+
+### Run single feature test
+
+The single feature tests can also be run against the different storage backends. Therefore multiple make targets with the schema test-<test source>-feature-<storage backend> exists. For selecting a single feature test you have to add an additional `BEHAT_FEATURE=...` parameter when invoking the make command:
+
+```
+make -C tests/acceptance/docker test-ocis-feature-ocis BEHAT_FEATURE='tests/acceptance/features/apiOcisSpecific/apiAuthOcs-ocsDELETEAuth.feature
```
Comment (micbar) [inline docs/ocis/development/testing.md:51]: missing `'`

Comment (wkloucek) [reply docs/ocis/development/testing.md]: added :+1:

**Comment by refs** [comment]
- I think `make test-owncloud-feature-ocis-owncloud` should be `make test-owncloud-feature-owncloud-storage` to be consistent with ` make test-owncloud-feature-ocis-storage`
- On my machine I tried running `make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis` but got:

```console
~/code/owncloud/ocis acceptance_tests_in_docker ⇣
❯ make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis
ERROR: The Compose file './src/redis.yml' is invalid because:
Unsupported config option for services: 'redis'
make[1]: *** [testSuite] Error 1
make: *** [localApiTests-apiOcisSpecific-ocis] Error 2
```

**Comment by refs** [comment]
@micbar did you try this on mac? did it work for you? I can't manage to get it working on my machine

File: tests/acceptance/docker/Makefile [hunk 2]
```diff
@@ -117,7 +117,7 @@ test-ocis-feature-owncloud-storage: ## test a ocis feature with oc10 storage, us
 
 .PHONY: test-oc10-feature-ocis-storage
 test-oc10-feature-ocis-storage: ## test a oC10 feature with oCIS storage, useage: make ... BEHAT_FEATURE='tests/acceptance/features/apiAuth/cors.feature'
-	@TEST_SOURCE=ocis \
+	@TEST_SOURCE=oc10 \
```
Comment (wkloucek) [inline tests/acceptance/docker/Makefile:120]: @micbar This fixed it. Please try again.

**Comment by wkloucek** [comment]
> * I think `make test-owncloud-feature-ocis-owncloud` should be `make test-owncloud-feature-owncloud-storage` to be consistent with ` make test-owncloud-feature-ocis-storage`

Good catch. Help and the actual command have been different.

> * On my machine I tried running `make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis` but got:
> 
> ```
> ~/code/owncloud/ocis acceptance_tests_in_docker ⇣
> ❯ make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis
> ERROR: The Compose file './src/redis.yml' is invalid because:
> Unsupported config option for services: 'redis'
> make[1]: *** [testSuite] Error 1
> make: *** [localApiTests-apiOcisSpecific-ocis] Error 2
> ```

Quick googling showed, that this error could happen indentations in the docker-compose YAML files are messed up. But the files seem to be ok. Could you try again?

**Comment by refs** [comment]
> > * I think `make test-owncloud-feature-ocis-owncloud` should be `make test-owncloud-feature-owncloud-storage` to be consistent with ` make test-owncloud-feature-ocis-storage`
> 
> Good catch. Help and the actual command have been different.
> 
> > * On my machine I tried running `make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis` but got:
> > 
> > ```
> > ~/code/owncloud/ocis acceptance_tests_in_docker ⇣
> > ❯ make -C tests/acceptance/docker localApiTests-apiOcisSpecific-ocis
> > ERROR: The Compose file './src/redis.yml' is invalid because:
> > Unsupported config option for services: 'redis'
> > make[1]: *** [testSuite] Error 1
> > make: *** [localApiTests-apiOcisSpecific-ocis] Error 2
> > ```
> 
> Quick googling showed, that this error could happen indentations in the docker-compose YAML files are messed up. But the files seem to be ok. Could you try again?

works for me now 🎉

**Review by refs — APPROVED** [review]

**Pull request merged**